src/include/storage/wal_replayer.h

```diff
@@ -15,15 +14,16 @@ namespace storage {
 
 class StorageManager;
 
+// Note: This class is not thread-safe.
 class WALReplayer {
 public:
     // This interface is used for recovery only. We always recover the disk files before
     // constructing the storageManager and catalog. So this specialized recovery constructor
     // doesn't take in storageManager and bufferManager.
-    WALReplayer(WAL* wal);
+    explicit WALReplayer(WAL* wal);
 
-    WALReplayer(WAL* wal, StorageManager* storageManager, BufferManager* bufferManager,
-        MemoryManager* memoryManager, catalog::Catalog* catalog, bool isCheckpoint);
+    WALReplayer(WAL* wal, StorageManager* storageManager, MemoryManager* memoryManager,
+        catalog::Catalog* catalog, bool isCheckpoint, bool isRecovering = false);
```

Comment:
Unless you are setting isRecovering to True in any case, don't expose isRecovering as a default argument.

Reply:
Done.